src/Internal.AspNetCore.HostingStartup.Sdk/build/Internal.AspNetCore.HostingStartup.Sdk.targets

```diff
@@ -0,0 +1,50 @@
+<!--
+  This file is automatically imported by NuGet into a user's project
+  when it targets a single framework, or in classic (pre 2017) csproj projects.
+-->
+
+<Project>
+
+  <Target Name="GenerateHostingStartupDeps" BeforeTargets="Build" Condition="'@(HostingStartupPackageReference->Count())' != '0'">
+    <PropertyGroup>
+      <_TemplatesDirectory>$(MSBuildThisFileDirectory)..\content\</_TemplatesDirectory>
+      <_DepsOutputDirectory>$(MSBuildProjectDirectory)\$(BaseIntermediateOutputPath)</_DepsOutputDirectory>
+      <_WorkingDirectory>$(_DepsOutputDirectory)\depswork</_WorkingDirectory>
+      <_BasePackagePath>content\additionaldeps\</_BasePackagePath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_TemplateFiles Include="$(MSBuildThisFileDirectory)..\content\HostingStartup\*.*" />
+      <_HostingStartupPackageReference
+        Include="%(HostingStartupPackageReference.Identity)"
+        Version="%(HostingStartupPackageReference.Version)"
+        WorkingDirectory="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)"
+        Project="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\HostingStartup.csproj"
+        DepsFile="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\HostingStartup.deps.json"
+        DllPath="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\%(HostingStartupPackageReference.Identity).dll"
+        TrimmedDepsFile="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\%(HostingStartupPackageReference.Identity).deps.json"
+        PackagePath="%(HostingStartupPackageReference.Identity)\shared\Microsoft.NETCore.App\$(HostingStartupRuntimeFrameworkVersion)\"
+         />
+    </ItemGroup>
+
+    <RemoveDir Directories="%(_HostingStartupPackageReference.WorkingDirectory)" />
+
+    <Copy SourceFiles="@(_TemplateFiles)" DestinationFolder="%(_HostingStartupPackageReference.WorkingDirectory)" />
+
+    <MSBuild Projects="%(_HostingStartupPackageReference.Project)" Targets="Restore" />
+
+    <MSBuild Projects="%(_HostingStartupPackageReference.Project)"
+             Targets="Restore;Clean;Publish"
+             Properties="PublishDir=%(_HostingStartupPackageReference.WorkingDirectory);HostingStartupPackageName=%(_HostingStartupPackageReference.Identity);HostingStartupPackageVersion=%(_HostingStartupPackageReference.Version)" />
+
+    <Copy SourceFiles="%(_HostingStartupPackageReference.DepsFile)" DestinationFiles="%(_HostingStartupPackageReference.TrimmedDepsFile)" />
+
+    <RepoTasks.TrimDeps DepsFiles="%(_HostingStartupPackageReference.TrimmedDepsFile)" />
+
+    <ItemGroup>
+      <Content Include="%(_HostingStartupPackageReference.TrimmedDepsFile)" PackagePath="%(_HostingStartupPackageReference.PackagePath)" />
```

Comment:
May not work when calling `dotnet pack --no-build` and in multi-targeting scenarios. Targets that add custom content to packages should add items to the `BuildOutputInPackage` item group and chain into TargetsForTfmSpecificBuildOutput.

cref https://github.com/NuGet/NuGet.Client/pull/1255

Example:
```xml
<PropertyGroup>
  <TargetsForTfmSpecificBuildOutput>$(TargetsForTfmSpecificBuildOutput);MyGenTarget</TargetsForTfmSpecificBuildOutput>
</PropertyGroup>

<Target Name="MyGenTarget">
 <!-- do stuff-->
  <ItemGroup>
     <BuildOutputInPackage Include="myfile.txt" />
   </ItemGroup>
</Target>
```

Reply:
Problem with `BuildOutputInPackage` is that it put's files into `lib` subdirectory even if `TargetPath` is set. So I have to set it to `TargetPath="..\..\$(_BasePackagePath)"`

Reply:
I think I'll stick with `Content` for now and see if we have any issues.

Reply:
Hmm, what about <TfmSpecificPackageFile> + TargetsForTfmSpecificContentInPackage?

https://github.com/NuGet/NuGet.Client/blob/4.3.0.4339/src/NuGet.Core/NuGet.Build.Tasks.Pack/Pack.targets#L275-L278

Also, you might consider simplifying this by just using a nuspec file. I've found this is the only way to do anything custom.

cc @rohit21agrawal